src/Microsoft.DotNet.XHarness.CLI/CommandArguments/Android/AndroidRunCommandArguments.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Mono.Options;
+
+namespace Microsoft.DotNet.XHarness.CLI.CommandArguments.Android
+{
+    internal class AndroidRunCommandArguments : TestCommandArguments
+    {
+        private string? _packageName;
+
+        /// <summary>
+        /// If specified, attempt to run instrumentation with this name instead of the default for the supplied APK.
+        /// If a given package has multiple instrumentations, failing to specify this may cause execution failure.
+        /// </summary>
+        public string? InstrumentationName { get; set; }
+
+        public string PackageName
+        {
+            get => _packageName ?? throw new ArgumentException("Package name not specified");
+            set => _packageName = value;
+        }
+
+        /// <summary>
+        /// Folder to copy off for output of executing the specified APK
+        /// </summary>
+        public string? DeviceOutputFolder { get; set; }
+
+        public Dictionary<string, string> InstrumentationArguments { get; } = new Dictionary<string, string>();
+
+        /// <summary>
+        /// Exit code returned by the instrumentation for a successful run. Defaults to 0.
+        /// </summary>
+        public int ExpectedExitCode { get; set; } = (int)Common.CLI.ExitCode.SUCCESS;
+
+        protected override OptionSet GetTestCommandOptions() => new OptionSet
+        {
+            { "device-out-folder=|dev-out=", "If specified, copy this folder recursively off the device to the path specified by the output directory",
+                v => DeviceOutputFolder = RootPath(v)
+            },
+            { "instrumentation:|i:", "If specified, attempt to run instrumentation with this name instead of the default for the supplied APK.",
+                v => InstrumentationName = v
+            },
+            { "expected-exit-code=", "If specified, sets the expected exit code for a successful instrumentation run.",
+                v => {
+                    if (int.TryParse(v, out var number))
+                    {
+                        ExpectedExitCode = number;
```

Comment:
Can't we do that in the ExpectedExitCode as the out already?

Reply:
No, it's property, it cannot be used as an out parameter